cfme/tests/cloud/test_cloud_proxy.py

```diff
@@ -0,0 +1,153 @@
+import pytest
+
+import cfme.fixtures.pytest_selenium as sel
+from cfme import test_requirements
+from cfme.cloud.provider import CloudProvider
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.web_ui import flash, form_buttons
+from utils import testgen, error
+from utils.conf import cfme_data, credentials
+from utils.log import logger
+from utils.appliance.implementations.ui import navigate_to
+
+pytestmark = [
+    pytest.mark.tier(1),
+    test_requirements.proxy,
+    pytest.mark.usefixtures('uses_cloud_providers')
+]
+pytest_generate_tests = testgen.generate([CloudProvider], scope="module")
+
+
+@pytest.mark.uncollectif(lambda provider: provider.one_of(OpenStackProvider))
```

Comment:
All tests are uncollected against OpenStackProvider, you should just generate on a ProviderFilter that excludes OpenStackProvider instead of generating for all CloudProvider and then uncollecting.

Example: https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/infrastructure/test_timelines.py#L23